src/support.js

```diff
@@ -435,9 +435,25 @@ function loadWebAssemblyModule(binary, flags) {
         if (prop.startsWith('invoke_')) {
           // A missing invoke, i.e., an invoke for a function type
           // present in the dynamic library but not in the main JS,
-          // and the dynamic library cannot provide JS for it. Use
-          // the generic "X" invoke for it.
-          return env[prop] = invoke_X;
+          // and the dynamic library cannot provide JS for it. Generate
+          // a closure for it.
+          var dynCallName = 'dynCall_' + prop.slice(7);
+          env[prop] = function() {
+            var sp = stackSave();
+            try {
+              var args = Array.prototype.slice.call(arguments);
+              if (Module.hasOwnProperty(dynCallName)) {
+                return Module[dynCallName].apply(null, args);
```

Comment:
For code size it might be nice to determine at compile time which of the two paths here we need. However we are refactoring that atm so it would not be convenient. Please add a comment though, maybe something like `// TODO: emit just the one necessary code path`.

Reply:
This seemed easy enough to do, so I just moved the `if` outside of the inner function.  If there is a `dynCall_XXX` it will use that, otherwise it falls back to `invoke_X` (the existing behavior).